tidy3d/plugins/design/design.py

```diff
@@ -45,39 +45,81 @@ class DesignSpace(Tidy3dBaseModel):
 
     name: str = pd.Field(None, title="Name", description="Optional name for the design space.")
 
+    task_name: str = pd.Field(
+        None,
+        title="Task Name",
+        description="Task name assigned to tasks along with a simulation counter. Only used when pre-post functions are supplied.",
+    )
+
+    path_dir: str = pd.Field(
+        ".",
+        title="Path Directory",
+        description="Directory for files to output to. Only used when pre-post functions are supplied.",
+    )
+
+    folder_name: str = pd.Field(
+        "default",
+        title="Folder Name",
+        description="Folder path where the simulation will be uploaded in the Tidy3D Workspace. Will use 'default' if no path is set.",
+    )
+
+    _dict_name: str = pd.PrivateAttr(default=None)
+
+    _task_names: list = pd.PrivateAttr(default=[])
+
+    _name_generator: Generator = pd.PrivateAttr(default=None)
+
     @cached_property
     def dims(self) -> Tuple[str]:
         """dimensions defined by the design parameter names."""
         return tuple(param.name for param in self.parameters)
 
-    @cached_property
-    def design_parameter_dict(self) -> Dict[str, ParameterType]:
-        """Mapping of design parameter name to design parameter."""
-        return dict(zip(self.dims, self.parameters))
-
     def _package_run_results(
         self,
         fn_args: Dict[str, tuple],
         fn_values: List[Any],
         fn_source: str,
         task_ids: Tuple[str] = None,
-        batch_data: BatchData = None,
+        aux_values: List[Any] = None,
     ) -> Result:
         """How to package results from ``method.run`` and ``method.run_batch``"""
 
-        fn_args_coords = tuple(fn_args.values())
+        fn_args_coords = tuple(
+            [[arg_dict[key] for arg_dict in fn_args] for key in fn_args[0].keys()]
```

Comment:
confused, so is `fn_args` no longer a `Dict[str, tuple]` like in the type annotation?

Reply:
Yes sorry forgot to update the typing. I'd made the change as all the optimisers produce inputs to the fitness function in the style of `dict[str, Any]`, and it didn't really make a difference to the samplers.